feat(tutorial): proxy-protocol-v2 ingress-nginx configuration (#1232)
scaleway/docs-content: tutorials/proxy-protocol-v2-load-balancer/index.mdx

    apache2ctl restart
    ```

## Configuring Proxy protocol for `ingress-nginx`

> 1. You'll need to add the two annotations below to your ingress controller `service` :

Reviewer: ```suggestion
1. Add the two annotations below to your ingress controller `service` :
```

    - [`service.beta.kubernetes.io/scw-loadbalancer-proxy-protocol-v2: "true"`](https://www.scaleway.com/en/docs/containers/kubernetes/api-cli/using-load-balancer-annotations/#service.beta.kubernetes.ioscw-loadbalancer-proxy-protocol-v2)
    - [`service.beta.kubernetes.io/scw-loadbalancer-use-hostname: "true"`](https://www.scaleway.com/en/docs/containers/kubernetes/api-cli/using-load-balancer-annotations/#service.beta.kubernetes.ioscw-loadbalancer-use-hostname)
   
    ```bash
    kubectl patch -n ingress-nginx services ingress-nginx-controller \
    -p '{"metadata":{"annotations":{"service.beta.kubernetes.io/scw-loadbalancer-proxy-protocol-v2": "true", "service.beta.kubernetes.io/scw-loadbalancer-use-hostname": "true"}}}'
    ```

> 2. And add the below configuration to the ingress controller `configmap` :

Reviewer: ```suggestion
2. Add the below configuration to the ingress controller `configmap` :
```
```suggestion
2. And add the below configuration to the ingress controller `configmap` :
```

    - [`use-forwarded-headers: "true"`](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-forwarded-headers)
    - [`compute-full-forwarded-for: "true"`](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#compute-full-forwarded-for)
    - [`use-proxy-protocol: "true"`](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-proxy-protocol)
    
    ```bash
    kubectl patch -n ingress-nginx configmaps ingress-nginx-controller \
    -p '{"data":{"use-forwarded-headers": "true", "compute-full-forwarded-for": "true", "use-proxy-protocol": "true"}}'
    ```

### Helm

> If you're using the [`ingress-nginx`](https://kubernetes.github.io/ingress-nginx/deploy/) Helm chart, here's a custom values file that will automatically create a load balancer resource and configure it to use `proxy-protocol-v2`, and add the required ingress controller configuration to support this new communication protocol :

Reviewer: ```suggestion
If you're using the [`ingress-nginx`](https://kubernetes.github.io/ingress-nginx/deploy/) Helm chart, find below a custom values file that will automatically create a load balancer resource and configure it to use `proxy-protocol-v2`, and add the required ingress controller configuration to support this new communication protocol :
```

1. Create the below file (e.g. `ingress-nginx-scw.yaml`)
    ```yaml
    controller:
    kind: DaemonSet
    service:
        # Your LB IP here if you want (optional)
        loadBalancerIP: "X.X.X.X"
        externalTrafficPolicy: "Local"
        annotations:
        service.beta.kubernetes.io/scw-loadbalancer-proxy-protocol-v2: "true"
        service.beta.kubernetes.io/scw-loadbalancer-use-hostname: "true"
    config:
        use-forwarded-headers: "true"
        compute-full-forwarded-for: "true"
        use-proxy-protocol: "true"
    ```

> 2. To apply the above on a new or an existing `ingress-nginx` installation :

Reviewer: ```suggestion
2. Apply the above, on a new or an existing `ingress-nginx` installation :
```

<Message type="important">
  Once ProxyProtocol is enabled the Apache virtual host will no longer accept direct connections.
</Message>

## Configuring Proxy protocol for `ingress-nginx`

### Deployed through `kubectl`

> 1. Add the two annotations below to your ingress controller `service` :

Reviewer 2: ```suggestion
1. Add the two annotations below to your ingress controller `service`:
```

>     - [`service.beta.kubernetes.io/scw-loadbalancer-proxy-protocol-v2: "true"`](https://www.scaleway.com/en/docs/containers/kubernetes/api-cli/using-load-balancer-annotations/#service.beta.kubernetes.ioscw-loadbalancer-proxy-protocol-v2)

Reviewer 2: ```suggestion
    - [`service.beta.kubernetes.io/scw-loadbalancer-proxy-protocol-v2: "true"`](/containers/kubernetes/api-cli/using-load-balancer-annotations/#service.beta.kubernetes.ioscw-loadbalancer-proxy-protocol-v2)
```

>     - [`service.beta.kubernetes.io/scw-loadbalancer-use-hostname: "true"`](https://www.scaleway.com/en/docs/containers/kubernetes/api-cli/using-load-balancer-annotations/#service.beta.kubernetes.ioscw-loadbalancer-use-hostname)

Reviewer 2: ```suggestion
    - [`service.beta.kubernetes.io/scw-loadbalancer-use-hostname: "true"`](/containers/kubernetes/api-cli/using-load-balancer-annotations/#service.beta.kubernetes.ioscw-loadbalancer-use-hostname)
```

   
    ```bash
    kubectl patch -n ingress-nginx services ingress-nginx-controller \
    -p '{"metadata":{"annotations":{"service.beta.kubernetes.io/scw-loadbalancer-proxy-protocol-v2": "true", "service.beta.kubernetes.io/scw-loadbalancer-use-hostname": "true"}}}'
    ```

> 2. Add the below configuration to the ingress controller `configmap` :

Reviewer 2: ```suggestion
2. Add the configuration below to the ingress controller `configmap`:
```

    - [`use-forwarded-headers: "true"`](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-forwarded-headers)
    - [`compute-full-forwarded-for: "true"`](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#compute-full-forwarded-for)
    - [`use-proxy-protocol: "true"`](https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-proxy-protocol)
    
    ```bash
    kubectl patch -n ingress-nginx configmaps ingress-nginx-controller \
    -p '{"data":{"use-forwarded-headers": "true", "compute-full-forwarded-for": "true", "use-proxy-protocol": "true"}}'
    ```

### Deployed through `helm`

> If you're using the [`ingress-nginx`](https://kubernetes.github.io/ingress-nginx/deploy/) Helm chart, find below a custom values file that will automatically create a load balancer resource and configure it to use `proxy-protocol-v2`, and add the required ingress controller configuration to support this new communication protocol :

Reviewer 2: ```suggestion
If you're using the [`ingress-nginx`](https://kubernetes.github.io/ingress-nginx/deploy/) Helm chart, find below a custom values file that will automatically create a load balancer resource and configure it to use `proxy-protocol-v2`, and add the required ingress controller configuration to support this new communication protocol:
```

> 1. Create the below file (e.g. `ingress-nginx-scw.yaml`)

Reviewer 2: ```suggestion
1. Create the file below (e.g. `ingress-nginx-scw.yaml`) using a text editor.
```

    ```yaml
    controller:
    kind: DaemonSet
    service:
        # Your LB IP here if you want (optional)
        loadBalancerIP: "X.X.X.X"
        externalTrafficPolicy: "Local"
        annotations:
        service.beta.kubernetes.io/scw-loadbalancer-proxy-protocol-v2: "true"
        service.beta.kubernetes.io/scw-loadbalancer-use-hostname: "true"
    config:
        use-forwarded-headers: "true"
        compute-full-forwarded-for: "true"
        use-proxy-protocol: "true"
    ```

> 2. Apply the above, on a new or an existing `ingress-nginx` installation :

Reviewer 2: ```suggestion
2. Apply the above, on a new or an existing `ingress-nginx` installation:
```
